subsys/modem/backends/modem_backend_uart_async.c

```diff
@@ -185,8 +185,8 @@ static int modem_backend_uart_async_transmit(void *data, const uint8_t *buf, siz
 	ret = uart_tx(backend->uart, backend->async.transmit_buf, bytes_to_transmit,
 		      CONFIG_MODEM_BACKEND_UART_ASYNC_TRANSMIT_TIMEOUT_MS * 1000L);
 
-	if (ret < 0) {
-		LOG_WRN("Failed to start async transmit");
+	if (ret != 0) {
+		LOG_ERR("Failed to start async transmit. (%d)", ret);
```

Comment:
it would be nice if such error messages are consistent so that the same buffer can be reused, which would result in smaller ROM size which I'm sure that an embedded device would appreciate, i.e.:

```c
LOG_ERR("%d: failed to %s %d bytes", ret, "async transmit", bytes_to_transmit);
```

or 

```c
LOG_ERR("%d: %s failed", ret, "async transmit");
```